dev/dll/Microsoft.UI.Xaml.vcxproj

```diff
@@ -596,89 +589,36 @@
-    </ItemGroup>
-    <ItemGroup Condition="'$(MiniWindowsSDKWinMDPath)'!=''">
-      <ProjectionWinMD Include="$(MiniWindowsSDKWinMDPath)\Windows.ApplicationModel.Calls.CallsVoipContract.winmd" />
-      <ProjectionWinMD Include="$(MiniWindowsSDKWinMDPath)\Windows.Devices.DevicesLowLevelContract.winmd" />
-      <ProjectionWinMD Include="$(MiniWindowsSDKWinMDPath)\Windows.Devices.Printers.PrintersContract.winmd" />
-      <ProjectionWinMD Include="$(MiniWindowsSDKWinMDPath)\Windows.Foundation.FoundationContract.winmd" />
-      <ProjectionWinMD Include="$(MiniWindowsSDKWinMDPath)\Windows.Foundation.UniversalApiContract.winmd" />
-      <ProjectionWinMD Include="$(MiniWindowsSDKWinMDPath)\Windows.Graphics.Printing3D.Printing3DContract.winmd" />
-      <ProjectionWinMD Include="$(MiniWindowsSDKWinMDPath)\Windows.Networking.Connectivity.WwanContract.winmd" />
-      <ProjectionWinMD Include="$(MiniWindowsSDKWinMDPath)\Windows.Services.Store.StoreContract.winmd" />
-      <ProjectionWinMD Include="$(MiniWindowsSDKWinMDPath)\Windows.System.Profile.ProfileHardwareTokenContract.winmd" />
-      <ProjectionWinMD Include="$(MiniWindowsSDKWinMDPath)\Windows.System.Profile.ProfileSharedModeContract.winmd" />
-      <ProjectionWinMD Include="$(MiniWindowsSDKWinMDPath)\Windows.UI.ViewManagement.ViewManagementViewScalingContract.winmd" />
-    </ItemGroup>
-    <ItemGroup>
-      <ComponentWinMD Include="$(OutDir)Microsoft.UI.Xaml.winmd" />
-      <ComponentWinMD Include="$(OutDir)Microsoft.UI.Private.winmd" />
-      <ComponentWinMD Include="$(OutDir)Microsoft.UI.Composition.winmd" />
-    </ItemGroup>
-    <Message Text="Generating Platform C++/WinRT headers ..." />
-    <Exec Command="$(CppWinRTExe) -overwrite @(ProjectionWinMD -&gt; '-in &quot;%(Identity)&quot;', ' ') -out $(IntermediateOutputPath)\CppWinRT\Platform" />
-    <Message Text="Generating Projection C++/WinRT headers ..." />
-    <Exec Command="$(CppWinRTExe) -component $(IntermediateOutputPath)\CppWinRT\Reference -overwrite @(ProjectionWinMD -&gt; '-ref &quot;%(Identity)&quot;', ' ') @(ComponentWinMD -&gt; '-in &quot;%(Identity)&quot;', ' ') -out $(IntermediateOutputPath)\CppWinRT\Component" />
-    <Message Text="Generating C++/WinRT headers complete" />
-    <ItemGroup>
-      <FileWrites Include="$(OutDir)Microsoft.UI.Xaml.winmd" />
-      <FileWrites Include="$(OutDir)Microsoft.UI.winmd" />
-      <FileWrites Include="$(OutDir)Microsoft.UI.Private.winmd" />
-      <FileWrites Include="$(OutDir)Microsoft.UI.Composition.winmd" />
-    </ItemGroup>
-  </Target>
-  <Target Name="CreateReferenceWinmds" AfterTargets="mdmerge" BeforeTargets="XamlMetadataCodeGenMUX">
+  <Target Name="CreateReferenceWinmds" BeforeTargets="XamlMetadataCodeGenMUX">
     <ItemGroup>
       <T4ParameterValues Include="ReferenceWinmds">
-        <Value>@(ProjectionWinMD);@(ComponentWinMD)</Value>
+        <Value>@(CppWinRTPlatformWinMDReferences)</Value>
```

Comment:
I wish msbuild had private artifacts.  I'll be sure never to rename this...

Reply:
Perhaps a naming convention like we have in C++/WinRT - cppwinrt_impl_paltformwinmdreferences...

Reply:
It's fine if you rename it -- I'll figure it out when I try to update and the build doesn't work anymore. :)

---
In reply to: [291289156](https://github.com/microsoft/microsoft-ui-xaml/pull/819#discussion_r291289156) [](ancestors = 291289156)